File: python-gdb-activities-20/activity19/gdb/tests/test_command_logging.py
# gdb/tests/test_command_logging.py
import unittest
import os
from gdb.domain.account_factory import AccountFactory
from gdb.command.deposit_command import DepositCommand
from gdb.command.withdraw_command import WithdrawCommand
from gdb.command.transfer_command import TransferCommand
from gdb.service.transfer_service import TransferService
from gdb.logging.transaction_log import TransactionLog

class TestCommandLogging(unittest.TestCase):
    def setUp(self):
        self.log_file = "test_transactions.log"
        self.log = TransactionLog(self.log_file)
        self.log.clear()
        self.acc1 = AccountFactory.create_account("SAVINGS", 1001, "Rajesh", 30, 50000.0)
        self.acc2 = AccountFactory.create_account("CURRENT", 1002, "Priya", 28, 50000.0)
        self.acc1.set_pin(1234)
        self.acc2.set_pin(5678)
        self.transfer_svc = TransferService()

    def tearDown(self):
        self.log.clear()

    def test_deposit_command(self):
        cmd = DepositCommand(self.acc1, 5000.0)
        cmd.execute()
        txn = cmd.get_transaction()
        self.assertIsNotNone(txn)
        self.log.log_transaction(txn)
        self.assertEqual(len(self.log.get_transactions()), 1)

    def test_withdraw_command(self):
        cmd = WithdrawCommand(self.acc1, 10000.0, 1234)
        cmd.execute()
        txn = cmd.get_transaction()
        self.assertIsNotNone(txn)
        self.log.log_transaction(txn)
        self.assertEqual(len(self.log.get_transactions()), 1)

    def test_transfer_command(self):
        cmd = TransferCommand(self.transfer_svc, self.acc1, self.acc2, 5000.0, 1234)
        cmd.execute()
        txn = cmd.get_transaction()
        self.assertIsNotNone(txn)
        self.log.log_transaction(txn)
        self.assertEqual(len(self.log.get_transactions()), 1)

    def test_file_persistence(self):
        cmd = DepositCommand(self.acc1, 3000.0)
        cmd.execute()
        self.log.log_transaction(cmd.get_transaction())
        
        new_log = TransactionLog(self.log_file)
        txns = new_log.load_from_file()
        self.assertEqual(len(txns), 1)

if __name__ == "__main__":
    unittest.main()

File: java-gdb-activities-20/activity17/src/com/gdb/tests/TestCommandLogging.java
package com.gdb.tests;

import com.gdb.command.*;
import com.gdb.domain.*;
import com.gdb.logging.TransactionLog;
import java.util.List;

public class TestCommandLogging {
    public static void main(String[] args) throws Exception {
        System.out.println("=".repeat(60));
        System.out.println("  ACTIVITY 17 — COMMAND PATTERN + FILE LOGGING");
        System.out.println("=".repeat(60));

        TransactionLog log = new TransactionLog();
        log.clear(); // start fresh

        // ============================================================
        // 📝 STEP 10: Create Test Accounts
        //
        // INSTRUCTIONS:
        //   1. acc1 = AccountFactory.createAccount("SAVINGS", 1001, "John Doe", 25, 15000);
        //   2. acc1.setPin(1234);
        //   3. acc2 = AccountFactory.createAccount("SAVINGS", 1002, "Jane Smith", 30, 10000);
        // ============================================================
        // TODO: create acc1, acc2 and set PIN on acc1

        // ============================================================
        // 📝 STEP 11: Execute and Log DepositCommand
        //
        // INSTRUCTIONS:
        //   1. Create DepositCommand depCmd = new DepositCommand(acc1, 5000);
        //   2. depCmd.execute();
        //   3. log.log(depCmd);
        //   4. Print depCmd.getTransaction();
        // ============================================================
        // TODO: execute and log DepositCommand

        // ============================================================
        // 📝 STEP 12: Execute and Log WithdrawCommand
        //
        // INSTRUCTIONS:
        //   1. Create WithdrawCommand wthCmd = new WithdrawCommand(acc1, 2000, 1234);
        //   2. wthCmd.execute();
        //   3. log.log(wthCmd);
        //   4. Print wthCmd.getTransaction();
        // ============================================================
        // TODO: execute and log WithdrawCommand

        // ============================================================
        // 📝 STEP 13: Execute and Log TransferCommand
        //
        // INSTRUCTIONS:
        //   1. Create TransferCommand trfCmd = new TransferCommand(acc1, acc2, 3000, 1234);
        //   2. trfCmd.execute();
        //   3. log.log(trfCmd);
        //   4. Print trfCmd.getTransaction();
        // ============================================================
        // TODO: execute and log TransferCommand

        // ============================================================
        // 📝 STEP 14: Read All Commands Back
        //
        // INSTRUCTIONS:
        //   1. List<TransactionCommand> history = log.readAll();
        //   2. Print history size and iterate printing each cmd.getTransaction();
        // ============================================================
        // TODO: read all logged commands and display audit trail

        // ============================================================
        // 📝 STEP 15: Verify File Persistence
        //
        // INSTRUCTIONS:
        //   1. TransactionLog freshLog = new TransactionLog();
        //   2. List<TransactionCommand> persisted = freshLog.readAll();
        //   3. Verify persisted.size() matches previous count.
        // ============================================================
        // TODO: verify persistence from independent log reader instance
    }
}
